### PR TITLE
Remove remainders of Accessor

### PR DESCRIPTION
**Pull request by bernhardmgruber**
- [x] Wait until #2061 is merged

### CODE REVIEW

File: include/alpaka/kernel/TaskKernelGenericSycl.hpp [hunk 1]
```diff
@@ -111,16 +110,6 @@ namespace alpaka::detail
...
 
-    template<typename TElem, typename TIdx, std::size_t TDim, typename TAccessModes>
-    inline auto require(
-        sycl::handler& cgh,
-        experimental::Accessor<SyclAccessor<TElem, DimInt<TDim>::value, TAccessModes>, TElem, TIdx, TDim, TAccessModes>
-            acc,
-        special)
-    {
-        cgh.require(acc.m_accessor);
-    }
```
Comment (bernhardmgruber) [inline include/alpaka/kernel/TaskKernelGenericSycl.hpp]: @j-stephan I don't quite understand what the replacement is here after the USM merge. Do we still need to require things anymore? And if yes, where do they come from?

File: include/alpaka/kernel/TaskKernelGenericSycl.hpp [hunk 2]
```diff
@@ -111,15 +110,6 @@ namespace alpaka::detail
...
 
-    template<typename TElem, typename TIdx, std::size_t TDim, typename TAccessModes>
-    inline auto require(
-        sycl::handler& cgh,
-        Accessor<SyclAccessor<TElem, DimInt<TDim>::value, TAccessModes>, TElem, TIdx, TDim, TAccessModes> acc,
-        special)
-    {
-        cgh.require(acc.m_accessor);
-    }
-
```
Comment (j-stephan) [inline include/alpaka/kernel/TaskKernelGenericSycl.hpp:122]: You can also remove the other `require` overloads I think. We needed them for SYCL accessors to global memory but we don't have those anymore.

**Comment by AuroraPerego** [comment]
Since this is affecting the SYCL back-end that is not checked in the CI yet, can you compile some tests for SYCL to test the changes?
Right now it's not compiling, there is an `#include "alpaka/mem/buf/sycl/Accessor.hpp"` in `include/alpaka/mem/buf/BufGenericSycl.hpp` that must be removed as well.

File: include/alpaka/kernel/TaskKernelGenericSycl.hpp (hunk 2 above)
Comment (AuroraPerego) [reply include/alpaka/kernel/TaskKernelGenericSycl.hpp:122]: yes, and therefore also lines 154-155

**Comment by AuroraPerego** [comment]
Since you are touching that file, also the `struct kernel` struct can be removed [here](https://github.com/alpaka-group/alpaka/pull/2062/files#diff-a73804fe988e37b9a98027f85583a10625529d1c6353e8ca7c0b3180462699e7L101-L104).
In short, everything in the `namespace alpaka::detail` is no longer used

File: include/alpaka/kernel/TaskKernelGenericSycl.hpp (hunk 2 above)
Comment (bernhardmgruber) [reply include/alpaka/kernel/TaskKernelGenericSycl.hpp:122]: thx, done!

**Comment by bernhardmgruber** [comment]
@AuroraPerego I applied your changes.

**Comment by j-stephan** [comment]
I added a missing `<cstdint>`. After adding this the tests compile successfully on my laptop (except for two which lead to a segmentation fault in the GPU compiler - I believe this is unrelated to the changes here).

**Review by j-stephan — APPROVED** [review]

**Pull request merged**